programs/ibc-attestor/src/adapter_impls/arbitrum.rs

```diff
@@ -1,77 +1,140 @@
-use std::time::Duration;
-
-use alloy::{consensus::Header as EthHeader, eips::BlockNumberOrTag};
+use alloy::{
+    consensus::BlockHeader,
+    eips::{BlockId, BlockNumberOrTag},
+};
 use alloy_network::Ethereum;
+use alloy_primitives::FixedBytes;
 use alloy_provider::{Provider, RootProvider};
+use attestor_packet_membership::Packets;
+use futures::{stream::FuturesUnordered, StreamExt};
+use ibc_eureka_solidity_types::ics26::{router::routerInstance, IICS26RouterMsgs::Packet};
 
 mod config;
 
 pub use config::ArbitrumClientConfig;
 
-use super::header::Header;
-use crate::adapter_client::{Adapter, AdapterError, Signable};
-
-/// Relevant chain peek options. For their Arbitrum
-/// interpretation see [these docs](https://docs.arbitrum.io/for-devs/troubleshooting-building#how-many-block-numbers-must-we-wait-for-in-arbitrum-before-we-can-confidently-state-that-the-transaction-has-reached-finality)
-enum PeekKind {
-    /// Most recent confirmed L2 block on ETH L1
-    Finalized,
-    /// Latest L2 block
-    Latest,
-}
+use crate::adapter_client::{
+    Adapter, AdapterError, UnsignedPacketAttestation, UnsignedStateAttestation,
+};
 
 #[derive(Debug)]
 pub struct ArbitrumClient {
     client: RootProvider,
+    router: routerInstance<RootProvider>,
 }
 
 impl ArbitrumClient {
     pub fn from_config(config: &ArbitrumClientConfig) -> Self {
         let client = RootProvider::<Ethereum>::new_http(config.url.parse().unwrap());
-        Self { client }
+        let address: FixedBytes<20> = config
+            .router_address
+            .clone()
+            .into_bytes()
+            .as_slice()
+            .try_into()
+            .unwrap();
+
+        let router = routerInstance::new(address.into(), client.clone());
+        Self { client, router }
     }
 
-    async fn get_block_by_number(&self, peek_kind: &PeekKind) -> Result<EthHeader, AdapterError> {
-        let kind = match peek_kind {
-            PeekKind::Finalized => BlockNumberOrTag::Finalized,
-            PeekKind::Latest => BlockNumberOrTag::Latest,
-        };
+    async fn get_latest_block_number(&self) -> Result<u64, AdapterError> {
+        self.client
+            // https://docs.arbitrum.io/for-devs/troubleshooting-building#how-many-block-numbers-must-we-wait-for-in-arbitrum-before-we-can-confidently-state-that-the-transaction-has-reached-finality
+            .get_block_by_number(BlockNumberOrTag::Latest)
+            .await
+            .map_err(|e| AdapterError::FinalizedBlockError(e.to_string()))?
```

Comment:
Slightly misleading error name, since we are not getting a finalized block? ;)

Reply:
But I guess this error is not from this PR, so we can just deal with that separately.